pkg/storage/unified/search/document.go

```diff
@@ -0,0 +1,171 @@
+				lookup := dashboard.CreateDatasourceLookup([]*dashboard.DatasourceQueryResult{
+					// TODO, query data sources
+				})
+				return &DashboardDocumentBuilder{
+					Namespace:        namespace,
+					DatasourceLookup: lookup,
+					Stats:            nil, // loaded in enterprise
+					Blob:             blob,
+				}, nil
+			},
+		},
+	}, nil
+}
+
+func (*defaultDocumentBuilder) BuildDocument(_ context.Context, key *resource.ResourceKey, rv int64, value []byte) (resource.IndexableDocument, error) {
+	tmp := &unstructured.Unstructured{}
+	err := tmp.UnmarshalJSON(value)
+	if err != nil {
+		return nil, err
+	}
+
+	obj, err := utils.MetaAccessor(tmp)
+	if err != nil {
+		return nil, err
+	}
+
+	doc := &StandardDocumentFields{}
+	doc.Load(key, rv, obj)
+
+	doc.Title = obj.FindTitle(doc.Name)
+	doc.ByteSize = len(value)
+
+	return doc, nil
+}
+
+// This is common across all resources
+type StandardDocumentFields struct {
+	// unique ID across everything (group+resource+namespace+name)
+	ID string `json:"id"`
+	RV int64  `json:"rv"`
+
+	Group     string `json:"group"`
+	Resource  string `json:"resource"`
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+
+	Folder      string   `json:"folder,omitempty"`
+	Title       string   `json:"title,omitempty"`
+	Description string   `json:"description,omitempty"`
+	Tags        []string `json:"tags,omitempty"`
+	ByteSize    int      `json:"byte_size,omitempty"`
+
+	// Standard k8s style labels
+	Labels map[string]string `json:"labels,omitempty"`
+
+	Created   time.Time  `json:"created,omitempty"`
+	CreatedBy string     `json:"created_by,omitempty"`
+	Updated   *time.Time `json:"updated,omitempty"`
+	UpdatedBy string     `json:"updated_by,omitempty"`
+
+	OriginName string `json:"origin_name,omitempty"`
+	OriginPath string `json:"origin_path,omitempty"`
+	OriginHash string `json:"origin_hash,omitempty"`
+	OriginTime int64  `json:"origin_time,omitempty"`
+}
+
+func (s *StandardDocumentFields) GetID() string {
+	return s.ID
+}
+
+// Load values from standard object
+func (s *StandardDocumentFields) Load(key *resource.ResourceKey, rv int64, obj utils.GrafanaMetaAccessor) {
+	s.ID = toID(key)
+	s.RV = rv
+	s.Labels = obj.GetLabels()
+
+	s.Group = key.Group
+	s.Resource = key.Resource
+	s.Namespace = key.Namespace
+	s.Name = key.Name
+
+	s.Folder = obj.GetFolder()
+	s.Created = obj.GetCreationTimestamp().Time.UTC()
+	s.CreatedBy = obj.GetCreatedBy()
+
+	ts, _ := obj.GetUpdatedTimestamp()
+	if ts != nil {
+		utc := ts.UTC()
+		s.Updated = &utc
+	}
+	s.UpdatedBy = obj.GetUpdatedBy()
+
+	origin, _ := obj.GetOriginInfo()
+	if origin != nil {
+		s.OriginName = origin.Name
+		s.OriginPath = origin.Path
+		s.OriginHash = origin.Hash
+		if origin.Timestamp != nil {
+			s.OriginTime = origin.Timestamp.UnixMilli()
+		}
+	}
+}
+
+func toID(key *resource.ResourceKey) string {
+	ns := key.Namespace
```

Comment:
curious why we need this.  isn't there a guid for the id?

Reply:
The GUID exists -- but i'm not sure if we should use it.  We are only now starting to enforce that one actually exists 🤣 

The meaning of the UID(GUID) is pretty weird internal k8s and really should not be used for much -- different deployments of the same resource are supposed to have different GUIDs // but actually be the same resource!  (for our implementation it is likely totally fine)

So... with that in mind it seems better to use something that semantically the key